spec/environment_spec.rb

```diff
@@ -551,4 +551,469 @@ def new_content(id)
+    consumer_cp.update_consumer({:environments => [{:name => @env.name}, {:name => envB.name}]})
+    ent = consumer_cp.list_entitlements()[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentA['id']}.1")).to be false
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be true
+  end
+
+  it 'should NOT re-gen consumer entitlement cert when environment is added where entitlement does
+    not provide content' do
+    consumer = @org_admin.register(random_string('testsystem'), :system, nil, {}, nil, nil, [], [],
+      [{ 'id' => @env['id']}])
+
+    expect(consumer['environments']).not_to be_nil
+
+    consumer_cp = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+    product = create_product
+    contentA = create_content
+    contentB = create_content
+    contentC = create_content
+    contentD = create_content
+    @cp.add_content_to_product(@owner['key'], product['id'], contentA['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentB['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], contentC['id'])
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentB['id'],}])
+    wait_for_job(job['id'], 15)
+    pool = @cp.create_pool(@owner['key'], product['id'])
+    ent = consumer_cp.consume_pool(pool['id'], {:quantity => 1})[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentA['id']}.1")).to be false
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be false
+
+    envB = @org_admin.create_environment(@owner['key'], 'envB', "envB", "For test systems only.")
+    job = @org_admin.promote_content(@env['id'],[{:contentId => contentD['id'],}])
+    wait_for_job(job['id'], 15)
+    consumer_cp.update_consumer({:environments => [{:name => @env.name}, {:name => envB.name}]})
+    ent = consumer_cp.list_entitlements()[0]
+    x509 = OpenSSL::X509::Certificate.new(ent['certificates'][0]['cert'])
+    extensions_hash = Hash[x509.extensions.collect { |ext| [ext.oid, ext.value] }]
+
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentB['id']}.1")).to be true
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentA['id']}.1")).to be false
+    expect(extensions_hash.has_key?("1.3.6.1.4.1.2312.9.2.#{contentC['id']}.1")).to be false
+  end
+
+  it 'should re-gen consumer entitlement cert when environment priority are reversed' do
```

Comment:
The name of this test is not entirely accurate. We should only regen the entitlement cert when the priority is reversed, AND at least one content got added/removed or originated from a different environment than before. Given that this test has 2 environments that each provide 1 unique content, the reversion of order should not result in the entitlement being regenerated (hence my other comment on the need to actually check the entitlement serial changed or not).

In addition to adding the entitlement serial check, we should change this test to something more useful like, there is only one content (ContentA), that is provided by both environments, and when the environment order is reversed, we assert that the entitlement did get regenerated (and also change it's name to something like `should re-gen consumer entitlement cert when environment priority are reversed and content originates from different environment`)